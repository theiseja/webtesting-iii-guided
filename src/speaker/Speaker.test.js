import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import renderer from 'react-test-renderer';

import Speaker from './Speaker';


describe('<Speaker />', () => {
    it('should match the snapshot', () => {
     const tree = renderer.create(<Speaker />); // boilerplate for renderer

     expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should match the snapshot with a message', () => {
      const tree = renderer.create(<Speaker message={'test message'} />); // returns a message

      expect(tree.toJSON()).toMatchSnapshot();
    });

  it('should call the speak function on click', () => {  // checks to make sure the speaker function works when you click the button
    const speakMock = jest.fn();
    const { getByText } = render(<Speaker speak={speakMock} />); // renders the Speaker component

    const speakButton = getByText(/speak/i); // finds the speak button

    fireEvent.click(speakButton);

    expect(speakMock).toHaveBeenCalled();
    });
});