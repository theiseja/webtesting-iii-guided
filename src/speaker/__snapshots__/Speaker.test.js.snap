// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Speaker /> should match the snapshot 1`] = `
Array [
  <button>
    Speak
  </button>,
  <div />,
]
`;

exports[`<Speaker /> should match the snapshot with a message 1`] = `
Array [
  <button>
    Speak
  </button>,
  <div>
    test message
  </div>,
]
`;
